### DISCUSSION TITLE
Part question, part idea

### DISCUSSION THREAD

**Discussion by nickrout** [sysadminsmedia/homebox#80] [General]
I wondered how to extend homebox to get new data fields. My particular use case is cataloguing an art collection. It would be useful to add fields such as medium (eg watercolor, oil, ceramic), artist, etc. I could probably use manufacturer for artist, but it feels like a hack.

**Comment by nickrout** [comment]
I just spotted the "Custom Fields"!!

**Comment by BoronFinder** [comment]
Seconded! I tried using the Custom Fields, but the experience would be a lot better if I didn't have to enter the custom field manually with every item that should have it.
Ideally, the custom field would be defined at the Label level, so when I create an Item with this Label, I would be presented with this custom field right upon entering edit mode.
E.g. if I want to inventorise a cable, I really want it to have a length, a box should have a color and optionally a holding capacity and so on.

Or, to spin this even further:
I would define at Label level which custom field I really want to be filled (making it recommended or even mandatory, so it would need to be filled at Item creation) or which is just strictly optional.
Or even: define the custom fields in a separate form (maybe complete with data type such as date, float or boolean, in case of numeric values maybe even with a unit such as centimeters, grams or whatever?), and add them to the Label without have to create the custom field every time.

**Comment by nobthenob** [comment]
As I have written [here](https://github.com/sysadminsmedia/homebox/discussions/1380#discussioncomment-16523120) I am also heavily using custom fields, and I am frustrated to add sets of custome fields manually again and again for every new item.
In addition to the ideas described above by BaronFinder, a kind of custom field configuration page would be nice, where you can select what custom fields should be always created with new items, or only in special context like the mentioned labels.
For optional custom fields they could e.g. be selected from a drop-down list, of all custom fields.

And once again: Custom fields should be searchable.

Thank you for your great work!!

**Reply by egdinger** [reply]
Have you looked into templates? They allow you to define custom fields that will always be added to an item when the template is selected, also allows you to add the same set of tags as well.

**Comment by nobthenob** [comment]
Yes, I tried Templates, but maybe I have not discovered the full potential. So far I mostly use the duplicate item function.

**Reply by egdinger** [reply]
I've created a template that I use for books, that has these custom fields defined (This is in the edit template dialog).
<img width="519" height="540" alt="image" src="https://github.com/user-attachments/assets/976339ee-4dc6-4bb5-83a6-a3e14525154b" />
And when I have the template selected on the create item dialog you can see it populate the tag, and it adds the custom fields (though not visible on this screen)
<img width="507" height="655" alt="image" src="https://github.com/user-attachments/assets/6bf864b3-cd64-4ece-a2a1-5fe3cea5a548" />

I guess I'm missing what you think is missing from templates?

**Reply by nobthenob** [reply]
Thank you for pushing my head again into using the template function.
So the workflow of creating multiple items of the same category could be like this:

1. create a suitable template
2. create a first item by using the template
3. create additional items from template or by duplication, depending on how similar the item data are, resp. what id the least effort.

I will definitely give it another try and create some templates for my typical item categories.
Thanks again!

**Reply by egdinger** [reply]
Yep that's the workflow. One other thing that can speed it up, you can put default values for the custom fields in the template and change them at any time. Right now I'm adding all my physical music, and I generally have them grouped by media type. So I've a set the default value for the Format custom field to Vinyl since that's what I'm going through.

It's a nice middle ground between duplicating an item and having to enter every field value if you're doing things that can be grouped.
